bgpd/bgp_evpn_mh.c

```diff
@@ -524,6 +524,100 @@ static int bgp_evpn_mh_route_delete(struct bgp *bgp, struct bgp_evpn_es *es,
 	return 0;
 }
 
+/*
+ * This function is called when the VNI RD changes.
+ * Advertise EAD/EVI local routes for this VNI
+ * in the global routing table with the new RD.
+ */
+int update_advertise_global_ead_evi_routes(struct bgp *bgp, struct bgpevpn *vpn)
+{
+	struct bgp_dest *rn, *global_rn;
+	struct bgp_path_info *pi, *global_pi;
+	struct attr *attr;
+	afi_t afi = AFI_L2VPN;
+	safi_t safi = SAFI_EVPN;
+	int route_changed = 0;
+	struct bgp_evpn_es *es;
+
+	/* Iterate over all local EAD/EVI routes for this VNI */
+	for (rn = bgp_table_top(vpn->route_table); rn;
+	     rn = bgp_route_next(rn)) {
+		struct prefix_evpn *evp = (struct prefix_evpn *)&rn->p;
+
+		if (evp->prefix.route_type != BGP_EVPN_AD_ROUTE)
+			continue;
+
+		/* Check if the EAD/EVI route is local */
+		for (pi = bgp_dest_get_bgp_path_info(rn); pi; pi = pi->next)
+			if (pi->peer == bgp->peer_self
+			    && pi->type == ZEBRA_ROUTE_BGP
+			    && pi->sub_type == BGP_ROUTE_STATIC)
+				break;
+		if (!pi)
+			continue;
```

Comment:
I don't see `pi` being set anyplace ... are we certain this will fall to being null/0 in the case where nothing is found? Just making sure...

Reply:
for (pi = bgp_dest_get_bgp_path_info(dest); pi; pi = pi->next)

bgp_dest_get_bgp_path_info will return NULL in case of a NULL pi. pi->next will be NULL as well if a valid pi does not exist. So, I think pi will be set to NULL always if a valid pi is not found.